### PR TITLE
Convert some libbeat python tests to go

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

Migrates some of the libbeat python integration tests to go.

## Why is it important?

Maintaining the python toolchain just for tests has lead to increased complexity and maintenance costs.  Moving to one testing toolchain lowers CI complexity and improves the developer experience.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally

```
cd libbeat
mage integTest
```

Runs both python and go integration tests

### CODE REVIEW

**Review by ycombinator — APPROVED** [review]
I ran `cd libbeat && mage integTest` and most of the tests passed.  The only one that failed was `TestFQDNEventSync` but that one failed for me on `main` too. 

Beyond that, I'm not sure how best to review this PR given it's size.

File: libbeat/tests/integration/framework.go [hunk 1]
```diff
@@ -294,6 +357,83 @@ func createTempDir(t *testing.T) string {
...
+	m := Meta{}
+	metaFile, err := os.Open(filepath.Join(b.TempDir(), "data", "meta.json"))
+	if err != nil {
+		return m, err
+	}
+	defer metaFile.Close()
+
+	metaBytes, _ := ioutil.ReadAll(metaFile)
+	json.Unmarshal(metaBytes, &m)
+	return m, nil
```
Comment (AndersonQ) [inline libbeat/tests/integration/framework.go:433]: [Blocker]
Why are you ignoring the errors? If either of them aren't `nil` the returned value cannot be used.

File: libbeat/tests/integration/framework.go [hunk 2]
```diff
@@ -303,41 +443,116 @@ func EnsureESIsRunning(t *testing.T) {
...
+
+	user := os.Getenv("ES_USER")
+	if user == "" {
+		user = "admin"
+	}
+
+	pass := os.Getenv("ES_PASS")
+	if pass == "" {
+		pass = "testing"
 	}
```
Comment (AndersonQ) [inline libbeat/tests/integration/framework.go:462]: Not the scope of this PR, but I believe the best would be to have a single function that loads those credentials. That way it'd be easy to know what can or need to be set and where they come from

File: libbeat/tests/integration/framework.go [hunk 3]
```diff
@@ -303,41 +443,116 @@ func EnsureESIsRunning(t *testing.T) {
 
 	esPort := os.Getenv("ES_PORT")
 	if esPort == "" {
-		esPort = "9200"
+		switch scheme {
+		case "http":
+			esPort = "9200"
+		case "https":
+			esPort = "9201"
+		}
```
Comment (AndersonQ) [inline libbeat/tests/integration/framework.go:451]: [NIT]
I see it'd break later on if the scheme is neither `http` nor `https`, but it'd be good to have a either a default or to return an error here.

**Review by AndersonQ — CHANGES_REQUESTED** [review]
I reviewed the `framework.go`, only the comments with _[Blocker]_ are indeed blockers, the others feel free to ignore if you do not agree or think ti's not worth applying them.

File: libbeat/tests/integration/framework.go [hunk 4]
```diff
@@ -123,6 +168,8 @@ func (b *BeatProc) Start(args ...string) {
 			// 4. Wait for the goroutine to finish, this helps ensuring
 			// no other Beat process was started
 			wg.Wait()
+		} else {
+			b.cmdMutex.Unlock()
```
[the remark this answers is not in the document]
Comment (leehinman) [reply libbeat/tests/integration/framework.go:172]: maybe. @belimawr has an Unlock() happening in the `if` part above.  I'd like them to comment to see if there was a reason not to just `defer` the Unlock().

File: libbeat/tests/integration/framework.go [hunk 5]
```diff
@@ -195,8 +259,6 @@ func (b *BeatProc) LogContains(s string) bool {
 
 // WaitForLogs waits for the specified string s to be present in the logs within
 // the given timeout duration and fails the test if s is not found.
-// msgAndArgs should be a format string and arguments that will be printed
-// if the logs are not found, providing additional context for debugging.
```
[the remark this answers is not in the document]
Comment (leehinman) [reply libbeat/tests/integration/framework.go:199]: thanks for catching

File: libbeat/tests/integration/framework.go (hunk 1 above)
Comment (leehinman) [reply libbeat/tests/integration/framework.go:433]: 🤦 fixed

File: libbeat/tests/integration/framework.go (hunk 3 above)
Comment (leehinman) [reply libbeat/tests/integration/framework.go:451]: added default

File: libbeat/tests/integration/framework.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (belimawr) [reply libbeat/tests/integration/framework.go:172]: Yes, there is a good reason, we need to release the mutex before calling `wg.Wait()` because the goroutine https://github.com/elastic/beats/blob/63e88307aa982712979822ca0262beaefef0de61/libbeat/tests/integration/framework.go#L140-L146 might call `b.startBeat()` and this will lock the mutex, so if it's not released before calling `wg.Wait()` we can end up on a deadlock.

We could improve the comment https://github.com/elastic/beats/blob/63e88307aa982712979822ca0262beaefef0de61/libbeat/tests/integration/framework.go#L165-L166 explaining this deadlock scenario.

[the remark this answers is not in the document]
Comment (leehinman) [reply libbeat/tests/integration/framework.go:175]: @AndersonQ I updated the comment to reflect info that @belimawr gave.  Can you give it another look?

File: libbeat/tests/integration/framework.go [hunk 6]
```diff
@@ -375,7 +376,7 @@ func EnsureESIsRunning(t *testing.T) {
 		// If you're reading this message, you probably forgot to start ES
 		// run `mage compose:Up` from Filebeat's folder to start all
 		// containers required for integration tests
-		t.Fatalf("cannot execute HTTP request to ES: %s", err)
+		t.Fatalf("cannot execute HTTP request to ES: %s,", err)
```
Comment (AndersonQ) [inline libbeat/tests/integration/framework.go:382]: Did you started to write something and forgot? Why did you add just a comma?

**Review by AndersonQ — APPROVED** [review]
